scripts/test_knowledge.py

```diff
@@ -0,0 +1,51 @@
+# Standard
+import operator
+
+# Third Party
+from datasets import Dataset
+from openai import OpenAI
+
+# First Party
+from src.instructlab.sdg import SDG
+from src.instructlab.sdg.default_flows import MMLUBenchFlow, SynthKnowledgeFlow
+from src.instructlab.sdg.pipeline import Pipeline
+
+# Please don't add you vLLM endpoint key here
+openai_api_key = "EMPTY"
+openai_api_base = "Add model endpoint here"
+
+client = OpenAI(
+    api_key=openai_api_key,
+    base_url=openai_api_base,
+)
+
+models = client.models.list()
+teacher_model = models.data[0].id
+router_model = models.data[1].id
+
+samples = [
+    {
+        "question_1": "what is the location of the tubal tonsils?",
+        "response_1": "The location of the tubal tonsils is the roof of the pharynx.",
+        "question_2": "How long does the adenoid grow?",
+        "task_description": "Teaching about human anatomy, specifically tonsils",
+        "response_2": "The adenoid grows until the age of 5, starts to shrink at the age of 7 and becomes small in adulthood.",
+        "question_3": "What is the immune systems first line of defense against ingested or inhaled foreign pathogens?",
+        "response_3": "The tonsils are the immune systems first line of defense.",
```

Comment:
Why is it numbered like this instead of being a list of q/a pairs?

Reply:
This depends on the prompt template. Essentially not all the seed questions provided in the taxonomy yaml goes in at a time. We will be using a sliding window (of size 3) approach to select the in context questions, then keep moving the window until it covers all the questions provided in the taxonomy yaml

Reply:
since that was the easiest way to format it into a prompt template, all the keys are directly injected into the prompt for consistency for example see below
```
For this {domain} domain here are some sample questions:
[Question]
{question_1}
[Answer]
{answer_1}
[End]
```

Reply:
got it -- so the sliding window bit needs to be handled in `generate_data()` then?  that works

Reply:
yep